### ISSUE TITLE
Refresh in-app-purchase App Store receipt for Mac

### ISSUE THREAD

**Issue by thomasdao** [electron/electron#21435]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

### Problem Description

Some users reported that they have purchased in app products but it's not reflected in the app. It happens in my app and several others as in this tickets #14945. The in-app receipt is not updated after user purchased the product.

### Proposed Solution
It would be helpful to have a way to refresh the in-app receipt using `SKReceiptRefreshRequest`.

### Alternatives Considered

If user purchases the product again, it works. I also checked with Apple and they said that "if the user deletes the app, and re-installs it from the App Store, the new appStoreReceipt should include the recent transaction". However these solutions are quite bad from user perspective. Having a way to refresh the receipt would make it a lot more reliable to implement in-app purchase.

**Comment by thomasdao** [comment]
Hi @codebytere, would it be possible to look at this feature request again if you have some time? 
Having an API to refresh receipt would really help us a lot, since we are receiving more and more complaints from our users that when they purchase the app, the Premium is not granted to them because the receipt is not updated.

We had created a standalone binary from Swift command line to refresh the receipt but always see the error `The operation couldn’t be completed. (SKErrorDomain error 0.)`. It's probably because StoreKit needs to be setup differently, but we have tried a lot of things but still can't get it to work. I hope this function can be added to Electron - it will help Electron apps to be more reliable. Thanks!

**Comment by thomasdao** [comment]
@kevinlinv I'm using `restoreCompletedTransactions` to refresh the receipt

**Comment by kevinlinv** [comment]
Ah, cool, thanks!

**Comment by ChrisChou-freeman** [comment]
> @kevinlinv I'm using `restoreCompletedTransactions` to refresh the receipt

Hi, I'm sorry to bother you. Can you please tell me if the restoreCompletedTransactions interface still has any effect. I have no effect in the macOS 14 system version in the electron 28 version.

**Comment by thomasdao** [comment]
@ChrisChou-freeman yes it works fine for me, I haven't had any issue with it for a long time.

**Comment by ChrisChou-freeman** [comment]
OK, thanks for response.
